lib/srv/discovery/discovery_test.go

```diff
@@ -2262,6 +2264,14 @@ func TestDiscoveryDatabase(t *testing.T) {
 			}
 
 			if tc.discoveryConfigStatusCheck != nil {
+				require.EventuallyWithT(t, func(c *assert.CollectT) {
+					dc, err := tlsServer.Auth().GetDiscoveryConfig(ctx, discoveryConfigName)
+					require.NoError(c, err)
+					require.Equal(c, tc.discoveryConfigStatusExpectedResources, int(dc.Status.DiscoveredResources))
+
+					tc.discoveryConfigStatusCheck(t, dc.Status)
```

Comment:
Are you sure you didn't mean to pass `c` here instead?

I prefer naming the `*assert.CollectT` `t` so that it's impossible to use the wrong one.